test/components/tableTests.ts

```diff
@@ -1,288 +1,112 @@
-    assert.deepEqual((<any> t)._rows, [], "the table rows is an empty list");
-    (<any> t)._padTableToSize(1, 1);
-    let rows = (<any> t)._rows;
-    let row = rows[0];
-    let firstComponent = row[0];
-    assert.lengthOf(rows, 1, "there is one row");
-    assert.lengthOf(row, 1, "the row has one element");
-    assert.isNull(firstComponent, "the row only has a null component");
-
-    (<any> t)._padTableToSize(5, 2);
-    assert.lengthOf(rows, 5, "there are five rows");
-    rows.forEach((r: Plottable.Component[]) => assert.lengthOf(r, 2, "there are two columns per row"));
-    assert.strictEqual(rows[0][0], firstComponent, "the first component is unchanged");
-  });
-
-  it("table constructor can take a list of lists of components", () => {
-    let c0 = new Plottable.Component();
-    let row1 = [null, c0];
-    let row2 = [new Plottable.Component(), null];
-    let table = new Plottable.Components.Table([row1, row2]);
-    assert.strictEqual((<any> table)._rows[0][1], c0, "the component is in the right spot");
-    let c1 = new Plottable.Component();
-    table.add(c1, 2, 2);
-    assert.strictEqual((<any> table)._rows[2][2], c1, "the inserted component went to the right spot");
-  });
-
-  describe("add()", () => {
-    it("adds Component and pads out other empty cells with null", () => {
-      let table = new Plottable.Components.Table();
-      let c1 = new Plottable.Component();
-      let c2 = new Plottable.Component();
-      table.add(c1, 0, 0);
-      table.add(c2, 1, 1);
-      let rows = (<any> table)._rows;
-      assert.lengthOf(rows, 2, "there are two rows");
-      assert.lengthOf(rows[0], 2, "two cols in first row");
-      assert.lengthOf(rows[1], 2, "two cols in second row");
-      assert.strictEqual(rows[0][0], c1, "first component added correctly");
-      assert.strictEqual(rows[1][1], c2, "second component added correctly");
-      assert.isNull(rows[0][1], "component at (0, 1) is null");
-      assert.isNull(rows[1][0], "component at (1, 0) is null");
-    });
-
-    it("adding a Component where one already exists throws an Error", () => {
-      let c1 = new Plottable.Component();
-      let t = new Plottable.Components.Table([[c1]]);
-      let c2 = new Plottable.Component();
-      assert.throws(() => t.add(c2, 0, 0), Error, "occupied");
-    });
-
-    it("adding null to a table cell should throw an error", () => {
-      let c1 = new Plottable.Component();
-      let t = new Plottable.Components.Table([[c1]]);
-
-      assert.throw(() => t.add(null, 0, 0), "Cannot add null to a table cell");
+  function assertTableRows(table: Plottable.Components.Table, expectedRows: Plottable.Component[][], message: string) {
+    expectedRows.forEach((row, rowIndex) => {
+      row.forEach((component, columnIndex) => {
+        assert.strictEqual(table.componentAt(rowIndex, columnIndex), component,
+          message + ` contains the correct entry at [${rowIndex}, ${columnIndex}]`);
```

Comment:
You have part of the string templated, might as well include `message` in the template